classes/orcid/OrcidManager.php

```diff
@@ -0,0 +1,365 @@
+    {
+        if (self::isGloballyConfigured()) {
+            $apiType = Application::get()->getRequest()->getSite()->getData(self::API_TYPE);
+        } else {
+            if ($context === null) {
+                $context = Application::get()->getRequest()->getContext();
+            }
+            $apiType = $context->getData(self::API_TYPE);
+        }
+        return in_array($apiType, [self::API_PUBLIC_PRODUCTION, self::API_MEMBER_PRODUCTION]) ? self::ORCID_URL : self::ORCID_URL_SANDBOX;
+    }
+
+    /**
+     * Gets the ORCID API URL, one of sandbox/production, member/public API URLs.
+     *
+     * Will first check if globally configured and prioritize site-level settings over context-level setting.
+     */
+    public static function getApiPath(?Context $context = null): string
+    {
+        if (self::isGloballyConfigured()) {
+            $apiType = Application::get()->getRequest()->getSite()->getData(self::API_TYPE);
+        } else {
+            if ($context === null) {
+                $context = Application::get()->getRequest()->getContext();
+            }
+            $apiType = $context->getData(self::API_TYPE);
+        }
+
+        return match ($apiType) {
+            self::API_PUBLIC_SANDBOX => self::ORCID_API_URL_PUBLIC_SANDBOX,
+            self::API_MEMBER_PRODUCTION => self::ORCID_API_URL_MEMBER,
+            self::API_MEMBER_SANDBOX => self::ORCID_API_URL_MEMBER_SANDBOX,
+            default => self::ORCID_API_URL_PUBLIC,
+        };
+    }
+
+    /**
+     * Returns whether the ORCID integration is set to use the sandbox domain.
+     *
+     * Will first check if globally configured and prioritize site-level settings over context-level setting.
+     */
+    public static function isSandbox(?Context $context = null): bool
+    {
+        if (self::isGloballyConfigured()) {
+            $apiType = Application::get()->getRequest()->getSite()->getData(self::API_TYPE);
+        } else {
+            if ($context === null) {
+                $context = Application::get()->getRequest()->getContext();
+            }
+            $apiType = $context->getData(self::API_TYPE);
+        }
+
+        return in_array($apiType, [self::API_PUBLIC_SANDBOX, self::API_MEMBER_SANDBOX]);
+    }
+
+    /**
+     * Constructs an ORCID OAuth URL with correct scope/API based on configured settings
+     *
+     * @param string $handlerMethod Previously: containting a valid method of the OrcidProfileHandler
+     * @param array $redirectParams Additional request parameters for the redirect URL
+     *
+     * @throws \Exception
+     */
+    public static function buildOAuthUrl(string $handlerMethod, array $redirectParams): string
+    {
+        $request = Application::get()->getRequest();
+        $context = $request->getContext();
+        if ($context === null) {
+            throw new \Exception('OAuth URLs should can only be made in a Context, not site wide');
```

Comment:
I think it's okay to throw the exception because a user should never encounter a case where this should happen, and I think the primary case where this would happen would be in development where I would want the exception thrown. It would be nice in the future to be able to get rid of the requirement of needing a context. How does that sound to you?